include/deal.II/base/mpi_noncontiguous_partitioner.templates.h

```diff
@@ -252,7 +259,8 @@ namespace Utilities
         requests.resize(send_ranks.size() + recv_ranks.size());
 
       if (this->buffers.size() != send_ptr.back() * sizeof(Number))
-        this->buffers.resize(send_ptr.back() * sizeof(Number), 0);
+        this->buffers.resize(this->temporary_storage_size() * sizeof(Number),
+                             0);
```

Comment:
Do we need to set the zero or leads relying on the default value to easier code?

Reply:
This is not needed. I have remove it.